libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -929,8 +937,10 @@ bool GCS_MAVLINK::should_send_message_in_delay_callback(const ap_message id) con
     // microseconds to return!
 
     switch (id) {
-    case MSG_HEARTBEAT:
     case MSG_NEXT_PARAM:
+        return is_high_latency_link ? false : true;
+    case MSG_HEARTBEAT:
+    case MSG_HIGH_LATENCY2:
```

Comment:
Is this a fast message to send?  i.e. won't do any really heavy calculations?

If this method returns true you're saying that you're not going to over-run the very small amount of time that the main loop might be sleeping right now.

Reply:
Removed.